### PR TITLE
 WAZO-3118 - Run pyupgrade to update to a more modern syntax and feature set

### PR DESCRIPTION
**Pull request by sopelj**

### CODE REVIEW

File: provd/devices/ident.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class RequestType(Enum):
     DHCP = 'dhcp'
 
 
-DHCPRequest = Dict[str, Union[str, Dict[str, str]]]
+DHCPRequest = dict[str, Union[str, dict[str, str]]]
```
Comment (fblackburn1) [inline provd/devices/ident.py:36]: Why changing this?

File: provd/operation.py [hunk 2]
```diff
@@ -44,7 +43,7 @@ def __init__(self, label=None, state=OIP_WAITING, current=None, end=None, sub_oi
 
 
 def format_oip(oip):
-    """Format an operation in progress to a string.
+    r"""Format an operation in progress to a string.
```
Comment (fblackburn1) [inline provd/operation.py:46]: Is it really necessary?

File: provd/persist/common.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
-# Copyright 2011-2022 The Wazo Authors  (see the AUTHORS file)
+# Copyright 2011-2023 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-"""Persistent storage interface for 'documents'.
+r"""Persistent storage interface for 'documents'.
```
Comment (fblackburn1) [inline provd/persist/common.py:4]: Same as above

File: provd/servers/tftp/packet.py [hunk 4]
```diff
@@ -12,8 +12,8 @@
 from typing import Dict, Union
 from provd.app import logger
 
-PacketOptions = Dict[bytes, bytes]
-Packet = Dict[str, Union[bytes, PacketOptions]]
+PacketOptions = dict[bytes, bytes]
+Packet = dict[str, Union[bytes, PacketOptions]]
```
Comment (fblackburn1) [inline provd/servers/tftp/packet.py:16]: same question as above

File: provd/servers/tftp/service.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
 
 from provd.servers.tftp.packet import ERR_FNF, Packet
 
-TFTPRequest = Dict[str, Union[str, Packet]]
+TFTPRequest = dict[str, Union[str, Packet]]
```
Comment (fblackburn1) [inline provd/servers/tftp/service.py:14]: same

File: provd/services.py [hunk 6]
```diff
@@ -22,7 +22,7 @@ class InvalidParameterError(Exception):
 
 
 class AbstractConfigurationService(metaclass=ABCMeta):
-    """Interface for a "configuration service".
+    r"""Interface for a "configuration service".
```
Comment (fblackburn1) [inline provd/services.py:25]: same

File: provd/devices/ident.py (hunk 1 above)
Comment (sopelj) [reply provd/devices/ident.py:36]: As of Python 3.9 literals can be used directly. This avoids needing to import the types from `typing`

File: provd/operation.py (hunk 2 above)
Comment (sopelj) [reply provd/operation.py:46]: In theory I think you are supposed to use raw strings when you have escape characters. This one might be ok, but the next one has `\w` so it definitely should.

File: provd/devices/ident.py (hunk 1 above)
Comment (fblackburn1) [reply provd/devices/ident.py:36]: Ah ok, then we have an unused import of `Dict`

File: provd/operation.py (hunk 2 above)
Comment (fblackburn1) [reply provd/operation.py:46]: ok

File: provd/devices/ident.py (hunk 1 above)
Comment (sopelj) [reply provd/devices/ident.py:36]: Oh, good catch. Thanks. I'll fix. Provd doesn't have any linters it seems.

**Review by fblackburn1 — APPROVED** [review]

**Pull request merged**